### PR TITLE
T244548 Set up preview counting for KaiOS app

### PR DESCRIPTION
**Pull request by hueitan**
Phabricator Link: https://phabricator.wikimedia.org/T244548

### Solution

Similar to `useTracking` code, reuse the `sentEvent` util for the data sending

### Note

Schema url : https://meta.wikimedia.org/wiki/Schema:VirtualPageView

### CODE REVIEW

File: src/hooks/useArticlePreviewTracking.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+) => {
+  if (!isInstrumentationEnabled()) {
+    return
+  }
+
+  if (page === undefined || source === undefined) {
+    return
+  }
+
+  const logInukaPageView = () => {
```
Comment (neilpquinn) [inline src/hooks/useArticlePreviewTracking.js:18]: We should probably update the name of this function :)

Comment (hueitan) [reply src/hooks/useArticlePreviewTracking.js:18]: Yes omg !!! Thanks @neilpquinn

File: src/hooks/useArticle.js [hunk 2]
```diff
@@ -3,7 +3,14 @@ import { useI18n } from 'hooks'
 import { getArticle, getArticleMediaList, getSuggestedArticles } from 'api'
 import { canonicalizeTitle } from 'utils'
 
+let currentArticle
```
Comment (stephanebisson) [inline src/hooks/useArticle.js:6]: Adding global state like that is not great. It took me a while to figure out the whole logic here. It's clever, and it works, but I would like us to explore more conventional (or "boring") options.

One such option is to give ArticlePreview the required info about the source on open
```
showArticlePreview({ title, lang, dir, <info about the current article> })
```

File: src/hooks/useArticlePreviewTracking.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+const SCHEMA_REV = 19883675
+
+export const useArticlePreviewTracking = (
+  page, source, language
+) => {
+  if (!isInstrumentationEnabled()) {
+    return
+  }
+
+  if (page === undefined || source === undefined) {
```
Comment (stephanebisson) [inline src/hooks/useArticlePreviewTracking.js:14]: When is this called without `page` or `source`?

**Review by stephanebisson — CHANGES_REQUESTED** [review]

File: src/hooks/useArticle.js (hunk 2 above)
Comment (hueitan) [reply src/hooks/useArticle.js:6]: I start with the conventional options, but then it's really bad =( but anyway, let me try again with it.

File: src/hooks/useArticlePreviewTracking.js (hunk 3 above)
Comment (hueitan) [reply src/hooks/useArticlePreviewTracking.js:14]: _Just in case `useArticle()` is being called before the `useArticle( x, y )`, to make sure it returns empty {} instead of error._

File: src/hooks/useArticle.js (hunk 2 above)
Comment (hueitan) [reply src/hooks/useArticle.js:6]: indeed, yes, `useArticleLinksNavigation` has `galleryItems` that also come from the source, we can have everything as the `source`, so it can pass to `showArticlePreview`

**Comment by hueitan** [comment]
Ready for review again @stephanebisson

**Review by stephanebisson — COMMENTED** [review]
It's annoying to propagate a bunch of article info through many layers but it makes the code very simple and predictable.

Will test tomorrow.

File: src/hooks/useArticlePreviewTracking.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      /* eslint-enable camelcase */
+    }
+
+    sendEvent(SCHEMA_NAME, SCHEMA_REV, language, event)
+  }
+
+  useLayoutEffect(() => {
+    if (!page) {
+      return
+    }
```
Comment (hueitan) [inline src/hooks/useArticlePreviewTracking.js:34]: so this is still needed here, the summary is a state and start with null until the api request is done

**Comment by hueitan** [comment]
> It's annoying to propagate a bunch of article info through many layers but it makes the code very simple and predictable.

Agree, I see the benefit of this solution now! it's easier to track and predict all the properties.

File: src/hooks/useArticlePreviewTracking.js (hunk 4 above)
Comment (stephanebisson) [reply src/hooks/useArticlePreviewTracking.js:34]: Why `useLayoutEffect` specifically instead of `useEffect`?

Layout effect is blocking and should only be used to prevent flickering on visual effects.
